src/main/resources/webroot/index.html

```diff
@@ -237,60 +148,109 @@ <h2>Create a new Vert.x application</h2>
         </div>
-        <div class="form-group row">
-          <label for="dependencies" class="col-sm-4 col-form-label">Dependencies</label>
+
+        <div class="form-group row" ng-show="vm.detailedOptionsCollapsed">
+          <label for="dependencies" class="col-sm-4 col-form-label">Dependencies
+            ({{vm.vertxProject.vertxDependencies.length}}/{{vm.availableVertxDependencies.length}})</label>
           <div class="col-sm-8">
             <input type="text" id="dependencies" class="form-control" placeholder="Web, MQTT, etc."
                    ng-model="vm.selectedDependency"
-                   uib-typeahead="dependency as dependency.name for dependency in vm.vertxDependencies | filter:$viewValue | limitTo:8"
+                   uib-typeahead="dependency as dependency.name for dependency in vm.availableVertxDependencies | filter:$viewValue | limitTo:8"
                    typeahead-on-select="vm.onDependencySelected($item, $model, $label, $event)">
           </div>
         </div>
+        <div id="detailedDependenciesAnchor"></div>
+        <div class="form-group row" id="detailedDependencies" ng-hide="vm.detailedOptionsCollapsed">
+          <div class="col-md-4">
+            <h3>Dependencies
+              ({{vm.vertxProject.vertxDependencies.length}}/{{vm.availableVertxDependencies.length}})</h3>
+            <ul class="nav nav-pills nav-stacked">
+              <li role="presentation" ng-repeat="cat in vm.stack" ng-class="{active:vm.selectedPanel === cat.code}"><a
+                ng-click="vm.selectedPanel = cat.code">{{cat.category}}</a></li>
+            </ul>
+          </div>
+          <div class="col-sm-8">
+            <div class="tab-content">
+              <div id="tab-{{cat.code}}" class="tab-pane fade"
+                   ng-class="{in: cat.code == vm.selectedPanel, active: cat.code == vm.selectedPanel}"
+                   ng-repeat="cat in vm.stack">
+                <h3>{{cat.category}}</h3>
+                <p>{{cat.description}}</p>
+                <div ng-repeat="dependency in cat.items">
+                  <label ng-class="{'text-muted': dependency.disabled}">
+                    <input
+                      type="checkbox"
+                      ng-disabled="dependency.disabled"
+                      value="{{dependency.name}}"
+                      ng-model="dependency.selected"
+                      ng-change="$parent.vm.onDependencySelected($item, dependency, $label, $event)">
+                    {{dependency.name}}
+                  </label>
+                  <p ng-class="{'text-muted': dependency.disabled}">{{dependency.description}}</p>
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
       </div>
     </div>
+    <div id="advancedAnchor"></div>
     <div class="row">
       <div class="col-sm-12">
         <p class="text-center">
+          <button type="button" class="btn btn-link btn-advanced" ng-click="vm.toggleDetailedOptions()">
```

Comment:
Instead of a detailed modules options link next to advanced options, could you add a link below the dependencies form label? For example: "Show dependencies panel" and "Hide dependencies panel".

Reply:
So something like this?

![2019-08-12_14-20-30](https://user-images.githubusercontent.com/6738956/62864645-9810b600-bd0c-11e9-942c-27c5e9d4dc95.gif)

Reply:
Exactly like this! :wink: